crates/oxc_linter/src/rules/jest/prefer_mock_promise_shorthand.rs

```diff
@@ -0,0 +1,441 @@
+                                return;
+                            };
+                            Self::report(
+                                is_once,
+                                property_span,
+                                Some(arrow_func.span),
+                                arg_expr,
+                                ctx,
+                            );
+                        }
+                        Statement::ExpressionStatement(expr_stmt) => {
+                            Self::report(
+                                is_once,
+                                property_span,
+                                Some(arrow_func.span),
+                                &expr_stmt.expression,
+                                ctx,
+                            );
+                        }
+                        _ => (),
+                    }
+                }
+                Expression::FunctionExpression(func_expr) => {
+                    if !func_expr.params.is_empty() {
+                        return;
+                    }
+                    let Some(func_body) = &func_expr.body else {
+                        return;
+                    };
+                    let Some(stmt) = func_body.statements.first() else {
+                        return;
+                    };
+                    let Statement::ReturnStatement(return_stmt) = stmt else {
+                        return;
+                    };
+                    let Some(arg_expr) = &return_stmt.argument else {
+                        return;
+                    };
+                    Self::report(is_once, property_span, Some(func_expr.span), arg_expr, ctx);
+                }
+                _ => (),
+            };
+        }
+    }
+}
+
+impl PreferMockPromiseShorthand {
+    fn report<'a>(
+        is_once: bool,
+        property_span: Span,
+        arg_span: Option<Span>,
+        arg_expr: &'a Expression<'a>,
+        ctx: &LintContext,
+    ) {
+        let Expression::CallExpression(call_expr) = arg_expr else {
+            return;
+        };
+        let arg_name = get_node_name(arg_expr);
+
+        if !arg_name.eq("Promise.resolve") && !arg_name.eq("Promise.reject") {
+            return;
+        }
+
+        let mock_promise_resolve =
+            if is_once { "mockResolvedValueOnce" } else { "mockResolvedValue" };
+        let mock_promise_reject =
+            if is_once { "mockRejectedValueOnce" } else { "mockRejectedValue" };
+        let prefer_name =
+            if arg_name.ends_with("reject") { mock_promise_reject } else { mock_promise_resolve };
+        let fix_span = arg_span.unwrap_or(call_expr.span);
+
+        ctx.diagnostic_with_fix(
+            UseMockShorthand(Atom::from(prefer_name).to_compact_str(), property_span),
+            || {
+                let content = Self::fix(prefer_name, call_expr, ctx);
+                Fix::new(content, Span::new(property_span.start, fix_span.end))
```

Comment:
Do we also need to [ban fixer](https://github.com/jest-community/eslint-plugin-jest/blob/013810650a4a253b2b8bd771ec32098c9e312fcf/src/rules/prefer-mock-promise-shorthand.ts#L76-L80) when argument's length exceeds 1?

Reply:
Nice~ It's not also need to ban the fixer, but it still also need to report. But I ignored it before.